crates/distribution-types/src/lib.rs

```diff
@@ -572,12 +612,18 @@ impl Name for Dist {
     }
 }
 
-impl DistributionMetadata for RegistryBuiltDist {
+impl DistributionMetadata for RegistryBuiltWheel {
     fn version_or_url(&self) -> VersionOrUrlRef {
         VersionOrUrlRef::Version(&self.filename.version)
     }
 }
 
+impl DistributionMetadata for RegistryBuiltDist {
+    fn version_or_url(&self) -> VersionOrUrlRef {
+        VersionOrUrlRef::Version(&self.best_wheel().filename.version)
```

Comment:
Can this be `self.best_wheel().version_or_url()`, to avoid re-encoding the implementation?